platform/core/PolyPodCoreSwift/Sources/PolyPodCoreSwift/FBObject.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+import PolyPodCore
+import FlatBuffers
+
+@dynamicMemberLookup
+/// A wrapper class used to manage the lifetime of the Flatbuffer memory.
+/// It will make sure to deallocate the memory once the model is no more used.
+public final class FBObject<FBModel> {
+    /// The bytes storage for the fbModel
+    private let rawPointer: UnsafeMutablePointer<UInt8>
+    /// The underlying flatbuffer model.
+    private let fbModel: FBModel
+    
+    init(_ rawPointer: UnsafeMutablePointer<UInt8>, _ fbModel: FBModel) {
+        self.rawPointer = rawPointer
+        self.fbModel = fbModel
+    }
+    
+    deinit {
+        free_bytes(rawPointer)
+    }
+    
+    public subscript<T>(dynamicMember keyPath: KeyPath<FBModel, T>) -> T {
+        fbModel[keyPath: keyPath]
```

Comment:
Does this check whether keyPath actually exists? Or is it implicit in the argument type and it will throw if it does not?

Reply:
This will never be the case, as the key paths in swift are autogenerated based on the `struct`(in this case FBModel) definition.